test/common/testlib.py

```diff
@@ -1273,13 +1273,13 @@ def failureHandler(exc_info, python_311=False):
                 (err_case, exc_info) = self.errors[-1]
                 if exc_info:
                     assert err_case == test_case
-                    setattr(test_case, "_ran_debug", True)
+                    test_case._ran_debug = True
```

Comment:
This seems like the sort of thing that might upset vulture...

Reply:
Ah, didn't have vulture installed, but now  I do! Can we not replace that with `ruff` soon :)

Reply:
```
not ok 8 /static-code/test-pyvulture
# test/common/testlib.py:2253: unused attribute '_testlib__nondestructive' (60% confidence)
# test/common/testlib.py:2264: unused attribute '_testlib__no_retry_when_changed' (60% confidence)
# test/common/testlib.py:2275: unused attribute '_testlib__todo' (60% confidence)
# test/common/testlib.py:2318: unused attribute '_testlib__timeout' (60% confidence)
```
Meh